src/runtime/ObjectStructure.cpp

```diff
@@ -315,9 +308,9 @@ ObjectStructure* ObjectStructureWithTransition::addProperty(const ObjectStructur
     ObjectStructure* newObjectStructure;
 
     size_t nextSize = m_properties.size() + 1;
-    if (nextSize > ESCARGOT_OBJECT_STRUCTURE_ACCESS_CACHE_BUILD_MIN_SIZE) {
-        newObjectStructure = new ObjectStructureWithMap(nameIsIndexString, hasSymbol, hasEnumerableProperty, m_properties, newItem);
-    } else if (nextSize > ESCARGOT_OBJECT_STRUCTURE_TRANSITION_MODE_MAX_SIZE || nameIsIndexString) {
+    // ObjectStructureWithTransition cannot directly convert to ObjectStructureWithMap by just adding one property
+    ASSERT(nextSize < ESCARGOT_OBJECT_STRUCTURE_ACCESS_CACHE_BUILD_MIN_SIZE);
+    if (nextSize > ESCARGOT_OBJECT_STRUCTURE_TRANSITION_MODE_MAX_SIZE || nameIsIndexString) {
```

Comment:
`ObjectStructureWithTransition` cannot directly convert to ObjectStructureWithMap by just adding one property,
So I removed it.